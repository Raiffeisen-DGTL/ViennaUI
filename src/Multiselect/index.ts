export { Multiselect, type MultiselectEvent, type MultiselectProps, type MultiselectScrollEvent } from './Multiselect';
export type { MultiselectLocalization } from './localization';
