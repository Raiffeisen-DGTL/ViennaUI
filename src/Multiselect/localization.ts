import { DSLocalizationProps, Localization } from '../Localization';

export interface MultiselectLocalizationMap {
    'ds.multiselect.empty'?: string;
    'ds.multiselect.extra'?: string;
}

export type MultiselectLocalization = Localization<MultiselectLocalizationMap, MultiselectLocalizationContext>;
interface Extra {
    count: number | string;
}

export type MultiselectLocalizationContext = Extra | undefined;

export type MultiselectLocalizationProps = DSLocalizationProps<
    MultiselectLocalizationMap,
    MultiselectLocalizationContext
>;

export const multiselectLocalizationMap: MultiselectLocalizationMap = {
    'ds.multiselect.empty': 'Нет элементов для отображения',
};

export const defaultMultiselectLocalization = (
    key: keyof MultiselectLocalizationMap,
    context: MultiselectLocalizationContext
) => {
    if (key === 'ds.multiselect.extra') {
        return `Еще ${context?.count}`;
    }

    return multiselectLocalizationMap[key];
};
