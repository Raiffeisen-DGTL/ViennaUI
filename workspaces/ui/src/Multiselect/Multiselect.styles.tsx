import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';
import { Wrapped } from '../Input/Input.styles';
import InputWrapper from '../Input/InputWrapper';

const multiselect = getPresets('select', {
    base: null,
    placeholder: null,
    placeholderDisabled: null,
    custom: null,
});

const current = getPresets('select.current', {
    size: 'size',
    custom: null,
});

const part = getPresets('select.part', {
    base: null,
    focus: null,
});

const chip = getPresets('select.chip', {
    base: null,
    size: 'size',
    text: null,
});

const chipIcon = getPresets('select.chip.icon', {
    base: null,
    size: 'size',
    hover: null,
});

const extra = getPresets('select.extra', {
    base: null,
    custom: null,
});

const wrapper = getPresets('select.wrapper', {
    size: 'size',
});

export const Part = styled.div<{ active?: boolean; right?: boolean }>`
    display: flex;
    align-items: center;
    justify-content: center;
    ${part.base}
    ${({ active }) =>
        active &&
        css`
            ${part.focus}
        `}
    &:hover {
        ${part.focus}
    }
`;

export const Current = styled.div<{ size?: 'xs' | 's' | 'm' | 'l' | 'xl' | 'xxl' }>`
    display: flex;
    width: 100%;
    align-items: center;
    height: 100%;
    display: flex;
    align-items: center;
    ${current.size}
    ${current.custom}
`;

export const Extra = styled.span`
    ${extra.base}
    margin-right: 30px;
    ${extra.custom}
`;

export const Box = styled.div<{ design?: 'outline' | 'material' }>`
    position: relative;
    ${multiselect.base}
    ${multiselect.custom}
`;

export const Chip = styled.div<{ size?: any }>`
    display: flex;
    ${chip.base}

    &:only-of-type {
        width: 100%;
    }

    span {
        ${chip.text}
        ${chip.size}
    }

    svg {
        ${chipIcon.base}
        ${chipIcon.size}
        &:hover {
            ${chipIcon.hover}
        }
    }
`;

export const Text = styled.span`
    flex: 1;
`;

export const Placeholder = styled.div<{ size?: any; disabled?: boolean }>`
    width: 100%;
    overflow: hidden;
    ${multiselect.placeholder}
    ${({ disabled }) => disabled && multiselect.placeholderDisabled}
`;

export const StyledInputWrapper = styled(InputWrapper as any)`
    ${Wrapped as any} {
        ${wrapper.size}Æ’
    }
`;
