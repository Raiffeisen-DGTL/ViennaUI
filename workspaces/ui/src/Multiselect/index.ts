export * from './Multiselect';
export type { MultiselectLocalization } from './localization';
