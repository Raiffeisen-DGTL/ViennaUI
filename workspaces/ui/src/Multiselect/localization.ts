import { DSLocalizationProps, Localization } from '../Localization';

export interface MultiselectLocalizationMap {
    'ds.multiselect.empty'?: string;
    'ds.multiselect.extra'?: string;
}

export type MultiselectLocalization = Localization<MultiselectLocalizationMap, MultiselectLocalizationCotext>;
interface Extra {
    count: number | string;
}

export type MultiselectLocalizationCotext = Extra | undefined;

export type MultiselectLocalizationProps = DSLocalizationProps<
    MultiselectLocalizationMap,
    MultiselectLocalizationCotext
>;

export const multiselectLocalizationMap: MultiselectLocalizationMap = {
    'ds.multiselect.empty': 'Нет элементов для отображения',
};

export const defaultMultiselectLocalization = (
    key: keyof MultiselectLocalizationMap,
    context: MultiselectLocalizationCotext
) => {
    if (key === 'ds.multiselect.extra') {
        return `Еще ${context?.count}`;
    }

    return multiselectLocalizationMap[key];
};
