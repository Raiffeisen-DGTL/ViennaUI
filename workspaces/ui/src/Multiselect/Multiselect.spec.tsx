import React from 'react';
import { Multiselect, MultiselectProps } from './Multiselect';

test('Multiselect /w design', async () => {
    const designs: MultiselectProps['design'][] = ['outline', 'material'];

    designs.forEach((design) => {
        const snap = snapshot.render(<Multiselect design={design} />);
        expect(snap).toMatchSnapshot();
    });
});

test('Multiselect /w size', async () => {
    const sizes: MultiselectProps['size'][] = ['xs', 's', 'm', 'l', 'xl', 'xxl'];

    sizes.forEach((size) => {
        const snap = snapshot.render(<Multiselect size={size} />);
        expect(snap).toMatchSnapshot();
    });
});

test('Multiselect /w invalid', () => {
    const snap = snapshot.mount(<Multiselect invalid />);

    expect(snap).toMatchSnapshot();
});

test('Multiselect /w disabled', () => {
    const snap = snapshot.mount(<Multiselect disabled />);

    expect(snap).toMatchSnapshot();
});

test('Multiselect /w onFocus()', () => {
    const fn = jest.fn();
    const snap = snapshot.mount(<Multiselect onFocus={fn} />);

    snap.simulate('focus');
    expect(fn).toBeCalled();
});

test('Multiselect /w onBlur()', () => {
    const fn = jest.fn();
    const snap = snapshot.mount(<Multiselect onBlur={fn} />);

    snap.simulate('blur');
    expect(fn).toBeCalled();
});

test('Multiselect w/ any props', () => {
    const props = {
        className: 'className',
        id: 'id',
        title: 'title',
    };

    const component = <Multiselect {...props} />;

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.shallow(component);
    for (const [key, value] of Object.entries(props)) {
        expect(wrapper.props()[key]).toEqual(value);
    }
});
