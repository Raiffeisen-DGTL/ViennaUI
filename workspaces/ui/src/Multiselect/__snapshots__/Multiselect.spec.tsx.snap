// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiselect /w design 1`] = `
<div
  aria-autocomplete="list"
  aria-disabled="false"
  aria-invalid="false"
  class="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe"
  role="combobox"
  tabindex="1"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 daAZxl Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
    target=".Multiselectstyles__Current-sc-1tat4gv-1"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Multiselectstyles__Current-sc-1tat4gv-1 dterzX"
      >
        <div
          class="Multiselectstyles__Placeholder-sc-1tat4gv-6 iKwBtQ"
        />
      </div>
    </div>
    <div
      class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
    >
      <div
        class="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`Multiselect /w design 2`] = `
<div
  aria-autocomplete="list"
  aria-disabled="false"
  aria-invalid="false"
  class="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe"
  role="combobox"
  tabindex="1"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 lkLMMk Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
    target=".Multiselectstyles__Current-sc-1tat4gv-1"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Multiselectstyles__Current-sc-1tat4gv-1 dterzX"
      >
        <div
          class="Multiselectstyles__Placeholder-sc-1tat4gv-6 iKwBtQ"
        />
      </div>
    </div>
    <div
      class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
    >
      <div
        class="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`Multiselect /w disabled 1`] = `
<Multiselect
  design="outline"
  disabled={true}
  maxListHeight={300}
  openWhenFocus={true}
  postfix={
    Object {
      "down": <SelectOpenDown
        color="currentColor"
        size="m"
      />,
      "up": <SelectHide
        color="currentColor"
        size="m"
      />,
    }
  }
  size="l"
  tabIndex={1}
>
  <Multiselectstyles__Box
    aria-autocomplete="list"
    aria-disabled={true}
    aria-invalid={false}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
    tabIndex={1}
  >
    <div
      aria-autocomplete="list"
      aria-disabled={true}
      aria-invalid={false}
      className="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      tabIndex={1}
    >
      <Multiselectstyles__StyledInputWrapper
        active={false}
        design="outline"
        disabled={true}
        size="l"
        target={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": e {
              "baseHash": 288996390,
              "baseStyle": undefined,
              "componentId": "Multiselectstyles__Current-sc-1tat4gv-1",
              "isStatic": false,
              "rules": Array [
                "display:flex;width:100%;align-items:center;height:100%;display:flex;align-items:center;",
                [Function],
                " ",
                [Function],
              ],
              "staticRulesId": "",
            },
            "displayName": "Multiselectstyles__Current",
            "foldedComponentIds": Array [],
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "Multiselectstyles__Current-sc-1tat4gv-1",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
      >
        <InputWrapper
          active={false}
          className="Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
          design="outline"
          disabled={true}
          size="l"
          target={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": e {
                "baseHash": 288996390,
                "baseStyle": undefined,
                "componentId": "Multiselectstyles__Current-sc-1tat4gv-1",
                "isStatic": false,
                "rules": Array [
                  "display:flex;width:100%;align-items:center;height:100%;display:flex;align-items:center;",
                  [Function],
                  " ",
                  [Function],
                ],
                "staticRulesId": "",
              },
              "displayName": "Multiselectstyles__Current",
              "foldedComponentIds": Array [],
              "render": [Function],
              "shouldForwardProp": undefined,
              "styledComponentId": "Multiselectstyles__Current-sc-1tat4gv-1",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
        >
          <Inputstyles__Wrapper
            active={false}
            className="Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
            design="outline"
            disabled={true}
            size="l"
            target={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": e {
                  "baseHash": 288996390,
                  "baseStyle": undefined,
                  "componentId": "Multiselectstyles__Current-sc-1tat4gv-1",
                  "isStatic": false,
                  "rules": Array [
                    "display:flex;width:100%;align-items:center;height:100%;display:flex;align-items:center;",
                    [Function],
                    " ",
                    [Function],
                  ],
                  "staticRulesId": "",
                },
                "displayName": "Multiselectstyles__Current",
                "foldedComponentIds": Array [],
                "render": [Function],
                "shouldForwardProp": undefined,
                "styledComponentId": "Multiselectstyles__Current-sc-1tat4gv-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
          >
            <label
              className="Inputstyles__Wrapper-hc8ttv-6 eFbvRo Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
              disabled={true}
              size="l"
              target={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": e {
                    "baseHash": 288996390,
                    "baseStyle": undefined,
                    "componentId": "Multiselectstyles__Current-sc-1tat4gv-1",
                    "isStatic": false,
                    "rules": Array [
                      "display:flex;width:100%;align-items:center;height:100%;display:flex;align-items:center;",
                      [Function],
                      " ",
                      [Function],
                    ],
                    "staticRulesId": "",
                  },
                  "displayName": "Multiselectstyles__Current",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "shouldForwardProp": undefined,
                  "styledComponentId": "Multiselectstyles__Current-sc-1tat4gv-1",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
            >
              <Inputstyles__Wrapped
                disabled={true}
              >
                <div
                  className="Inputstyles__Wrapped-hc8ttv-5 eqanMK"
                  disabled={true}
                >
                  <Multiselectstyles__Current
                    key=".1"
                    size="l"
                  >
                    <div
                      className="Multiselectstyles__Current-sc-1tat4gv-1 dterzX"
                      size="l"
                    >
                      <Multiselectstyles__Placeholder
                        disabled={true}
                      >
                        <div
                          className="Multiselectstyles__Placeholder-sc-1tat4gv-6 cjQYwt"
                          disabled={true}
                        />
                      </Multiselectstyles__Placeholder>
                    </div>
                  </Multiselectstyles__Current>
                </div>
              </Inputstyles__Wrapped>
              <Inputstyles__Postfix
                active={false}
                size="l"
              >
                <div
                  className="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
                  size="l"
                >
                  <Multiselectstyles__Part
                    active={false}
                    key=".2"
                    onMouseDown={[Function]}
                    right={true}
                  >
                    <div
                      className="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
                      onMouseDown={[Function]}
                    >
                      <SelectOpenDown
                        color="currentColor"
                        size="m"
                      >
                        <svg
                          fill="currentColor"
                          height={20}
                          viewBox="0 0 24 24"
                          width={20}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <defs />
                          <path
                            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
                          />
                        </svg>
                      </SelectOpenDown>
                    </div>
                  </Multiselectstyles__Part>
                </div>
              </Inputstyles__Postfix>
            </label>
          </Inputstyles__Wrapper>
        </InputWrapper>
      </Multiselectstyles__StyledInputWrapper>
    </div>
  </Multiselectstyles__Box>
</Multiselect>
`;

exports[`Multiselect /w invalid 1`] = `
<Multiselect
  design="outline"
  invalid={true}
  maxListHeight={300}
  openWhenFocus={true}
  postfix={
    Object {
      "down": <SelectOpenDown
        color="currentColor"
        size="m"
      />,
      "up": <SelectHide
        color="currentColor"
        size="m"
      />,
    }
  }
  size="l"
  tabIndex={1}
>
  <Multiselectstyles__Box
    aria-autocomplete="list"
    aria-disabled={false}
    aria-invalid={true}
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="combobox"
    tabIndex={1}
  >
    <div
      aria-autocomplete="list"
      aria-disabled={false}
      aria-invalid={true}
      className="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      tabIndex={1}
    >
      <Multiselectstyles__StyledInputWrapper
        active={false}
        design="outline"
        invalid={true}
        size="l"
        target={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": e {
              "baseHash": 288996390,
              "baseStyle": undefined,
              "componentId": "Multiselectstyles__Current-sc-1tat4gv-1",
              "isStatic": false,
              "rules": Array [
                "display:flex;width:100%;align-items:center;height:100%;display:flex;align-items:center;",
                [Function],
                " ",
                [Function],
              ],
              "staticRulesId": "",
            },
            "displayName": "Multiselectstyles__Current",
            "foldedComponentIds": Array [],
            "render": [Function],
            "shouldForwardProp": undefined,
            "styledComponentId": "Multiselectstyles__Current-sc-1tat4gv-1",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
      >
        <InputWrapper
          active={false}
          className="Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
          design="outline"
          invalid={true}
          size="l"
          target={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": e {
                "baseHash": 288996390,
                "baseStyle": undefined,
                "componentId": "Multiselectstyles__Current-sc-1tat4gv-1",
                "isStatic": false,
                "rules": Array [
                  "display:flex;width:100%;align-items:center;height:100%;display:flex;align-items:center;",
                  [Function],
                  " ",
                  [Function],
                ],
                "staticRulesId": "",
              },
              "displayName": "Multiselectstyles__Current",
              "foldedComponentIds": Array [],
              "render": [Function],
              "shouldForwardProp": undefined,
              "styledComponentId": "Multiselectstyles__Current-sc-1tat4gv-1",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
        >
          <Inputstyles__Wrapper
            active={false}
            className="Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
            design="outline"
            invalid={true}
            size="l"
            target={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": e {
                  "baseHash": 288996390,
                  "baseStyle": undefined,
                  "componentId": "Multiselectstyles__Current-sc-1tat4gv-1",
                  "isStatic": false,
                  "rules": Array [
                    "display:flex;width:100%;align-items:center;height:100%;display:flex;align-items:center;",
                    [Function],
                    " ",
                    [Function],
                  ],
                  "staticRulesId": "",
                },
                "displayName": "Multiselectstyles__Current",
                "foldedComponentIds": Array [],
                "render": [Function],
                "shouldForwardProp": undefined,
                "styledComponentId": "Multiselectstyles__Current-sc-1tat4gv-1",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
          >
            <label
              className="Inputstyles__Wrapper-hc8ttv-6 iaA-dFS Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
              size="l"
              target={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": e {
                    "baseHash": 288996390,
                    "baseStyle": undefined,
                    "componentId": "Multiselectstyles__Current-sc-1tat4gv-1",
                    "isStatic": false,
                    "rules": Array [
                      "display:flex;width:100%;align-items:center;height:100%;display:flex;align-items:center;",
                      [Function],
                      " ",
                      [Function],
                    ],
                    "staticRulesId": "",
                  },
                  "displayName": "Multiselectstyles__Current",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "shouldForwardProp": undefined,
                  "styledComponentId": "Multiselectstyles__Current-sc-1tat4gv-1",
                  "target": "div",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
            >
              <Inputstyles__Wrapped>
                <div
                  className="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
                >
                  <Multiselectstyles__Current
                    key=".1"
                    size="l"
                  >
                    <div
                      className="Multiselectstyles__Current-sc-1tat4gv-1 dterzX"
                      size="l"
                    >
                      <Multiselectstyles__Placeholder>
                        <div
                          className="Multiselectstyles__Placeholder-sc-1tat4gv-6 iKwBtQ"
                        />
                      </Multiselectstyles__Placeholder>
                    </div>
                  </Multiselectstyles__Current>
                </div>
              </Inputstyles__Wrapped>
              <Inputstyles__Postfix
                active={false}
                size="l"
              >
                <div
                  className="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
                  size="l"
                >
                  <Multiselectstyles__Part
                    active={false}
                    key=".2"
                    onMouseDown={[Function]}
                    right={true}
                  >
                    <div
                      className="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
                      onMouseDown={[Function]}
                    >
                      <SelectOpenDown
                        color="currentColor"
                        size="m"
                      >
                        <svg
                          fill="currentColor"
                          height={20}
                          viewBox="0 0 24 24"
                          width={20}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <defs />
                          <path
                            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
                          />
                        </svg>
                      </SelectOpenDown>
                    </div>
                  </Multiselectstyles__Part>
                </div>
              </Inputstyles__Postfix>
            </label>
          </Inputstyles__Wrapper>
        </InputWrapper>
      </Multiselectstyles__StyledInputWrapper>
    </div>
  </Multiselectstyles__Box>
</Multiselect>
`;

exports[`Multiselect /w size 1`] = `
<div
  aria-autocomplete="list"
  aria-disabled="false"
  aria-invalid="false"
  class="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe"
  role="combobox"
  tabindex="1"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 gPunPw Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 bVcpLM"
    target=".Multiselectstyles__Current-sc-1tat4gv-1"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Multiselectstyles__Current-sc-1tat4gv-1 hXXTHs"
      >
        <div
          class="Multiselectstyles__Placeholder-sc-1tat4gv-6 iKwBtQ"
        />
      </div>
    </div>
    <div
      class="Inputstyles__Postfix-hc8ttv-4 lldsPY"
    >
      <div
        class="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
      >
        <svg
          fill="currentColor"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`Multiselect /w size 2`] = `
<div
  aria-autocomplete="list"
  aria-disabled="false"
  aria-invalid="false"
  class="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe"
  role="combobox"
  tabindex="1"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 eablnU Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 aBqAw"
    target=".Multiselectstyles__Current-sc-1tat4gv-1"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Multiselectstyles__Current-sc-1tat4gv-1 dterzX"
      >
        <div
          class="Multiselectstyles__Placeholder-sc-1tat4gv-6 iKwBtQ"
        />
      </div>
    </div>
    <div
      class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
    >
      <div
        class="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`Multiselect /w size 3`] = `
<div
  aria-autocomplete="list"
  aria-disabled="false"
  aria-invalid="false"
  class="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe"
  role="combobox"
  tabindex="1"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 hmYQOg Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 aBqAw"
    target=".Multiselectstyles__Current-sc-1tat4gv-1"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Multiselectstyles__Current-sc-1tat4gv-1 dterzX"
      >
        <div
          class="Multiselectstyles__Placeholder-sc-1tat4gv-6 iKwBtQ"
        />
      </div>
    </div>
    <div
      class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
    >
      <div
        class="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`Multiselect /w size 4`] = `
<div
  aria-autocomplete="list"
  aria-disabled="false"
  aria-invalid="false"
  class="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe"
  role="combobox"
  tabindex="1"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 daAZxl Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
    target=".Multiselectstyles__Current-sc-1tat4gv-1"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Multiselectstyles__Current-sc-1tat4gv-1 dterzX"
      >
        <div
          class="Multiselectstyles__Placeholder-sc-1tat4gv-6 iKwBtQ"
        />
      </div>
    </div>
    <div
      class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
    >
      <div
        class="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`Multiselect /w size 5`] = `
<div
  aria-autocomplete="list"
  aria-disabled="false"
  aria-invalid="false"
  class="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe"
  role="combobox"
  tabindex="1"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 dcSDdd Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
    target=".Multiselectstyles__Current-sc-1tat4gv-1"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Multiselectstyles__Current-sc-1tat4gv-1 dUdgWZ"
      >
        <div
          class="Multiselectstyles__Placeholder-sc-1tat4gv-6 iKwBtQ"
        />
      </div>
    </div>
    <div
      class="Inputstyles__Postfix-hc8ttv-4 cxucT"
    >
      <div
        class="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`Multiselect /w size 6`] = `
<div
  aria-autocomplete="list"
  aria-disabled="false"
  aria-invalid="false"
  class="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe"
  role="combobox"
  tabindex="1"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 femZgC Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
    target=".Multiselectstyles__Current-sc-1tat4gv-1"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Multiselectstyles__Current-sc-1tat4gv-1 dUdgWZ"
      >
        <div
          class="Multiselectstyles__Placeholder-sc-1tat4gv-6 iKwBtQ"
        />
      </div>
    </div>
    <div
      class="Inputstyles__Postfix-hc8ttv-4 cxucT"
    >
      <div
        class="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;

exports[`Multiselect w/ any props 1`] = `
<div
  aria-autocomplete="list"
  aria-disabled="false"
  aria-invalid="false"
  class="Multiselectstyles__Box-sc-1tat4gv-3 kvEYoe className"
  id="id"
  role="combobox"
  tabindex="1"
  title="title"
>
  <label
    class="Inputstyles__Wrapper-hc8ttv-6 daAZxl Multiselectstyles__StyledInputWrapper-sc-1tat4gv-7 gKKySP"
    target=".Multiselectstyles__Current-sc-1tat4gv-1"
  >
    <div
      class="Inputstyles__Wrapped-hc8ttv-5 lhoBGY"
    >
      <div
        class="Multiselectstyles__Current-sc-1tat4gv-1 dterzX"
      >
        <div
          class="Multiselectstyles__Placeholder-sc-1tat4gv-6 iKwBtQ"
        />
      </div>
    </div>
    <div
      class="Inputstyles__Postfix-hc8ttv-4 dvGfwK"
    >
      <div
        class="Multiselectstyles__Part-sc-1tat4gv-0 cvCdgk"
      >
        <svg
          fill="currentColor"
          height="20"
          viewBox="0 0 24 24"
          width="20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <defs />
          <path
            d="M12 14.44l7.293-7.294 1.414 1.415-8 8a1 1 0 01-1.414 0l-8-8 1.414-1.415L12 14.44z"
          />
        </svg>
      </div>
    </div>
  </label>
</div>
`;
